torch/csrc/jit/codegen/cuda/arith.h

```diff
@@ -184,6 +186,18 @@ TORCH_CUDA_API TensorView* threshold(TensorView* in, Val* thresh, Val* value);
 TORCH_CUDA_API Val* clamp(Val* in, Val* min_val, Val* max_val);
 TORCH_CUDA_API TensorView* clamp(TensorView* in, Val* min_val, Val* max_val);
 
+// Internal operator for supporting backward graphs
+//
+// example:
+//   v1 = T1 [I0(10),I1(20),I2(30),I3(40)]
+//   v2 = sum_to(v1,{30,1}) ------> v2 = T2[I2,R3 (keep_dim)]
+//
+//  This operator will return v1* directly if sizes of v1 root domain
+//  is already the same as shape.
+TORCH_CUDA_API TensorView* sum_to(
+    TensorView* v1,
+    const std::vector<Int*>& shape);
```

Comment:
why not using std::vector<int> for the shape?

Reply:
I wasn't sure if the question was about using `const vector<Int*> &` or not using `const vector<int> &`

Regarding `const vector<>&` I just followed the style of reduction_op but would be happy to change and help refactor if other signature styles are encouraged.

As for `vector<Int*>`, this operator is internal so not user facing, and in the only case we need it the backward API gives symbolic integers for the shape.

Reply:
My bad, I forgot to escape the comment (and markdown eat the `<int>` part). I was askign about the latter - thanks for the explanation, it answers my question.